skcosmo/utils/_pcovr_utils.py

```diff
@@ -8,9 +8,89 @@
 from sklearn.utils.validation import check_is_fitted
 
 
-def check_lr_fit(regressor, X, y=None):
+def _check_coefs(fitted_regressor, X, y):
     r"""
-    Checks that an regressor is fitted, and if not,
+    Checks that the coefficients of a fitted
+    regression model is compatible with the shapes
+    of X and y
+
+    :param regressor: sklearn-style regressor
+    :type regressor: object
+    :param X: feature matrix with which to compare the
+        regression coefficients
+    :type X: array
+    :param y: target values with which to compare the
+        regression coefficients
+    :type y: array
+
+    """
+    if fitted_regressor.coef_.ndim != y.ndim:
+        raise ValueError(
+            "The target regressor has a shape incompatible "
+            "with the supplied target space"
+        )
+    elif fitted_regressor.coef_.ndim == 1:
+        if fitted_regressor.coef_.shape[0] != X.shape[1]:
+            raise ValueError(
+                "The target regressor has a shape incompatible "
+                "with the supplied feature space"
+            )
+    else:
+        if fitted_regressor.coef_.shape[0] != y.shape[1]:
+            raise ValueError(
+                "The target regressor has a shape incompatible "
+                "with the supplied target space"
+            )
+        elif fitted_regressor.coef_.shape[1] != X.shape[1]:
+            raise ValueError(
+                "The target regressor has a shape incompatible "
+                "with the supplied feature space"
+            )
+
+
+def _check_dual_coefs(fitted_regressor, K, y):
+    r"""
+    Checks that the dual coefficients of a fitted
+    regression model is compatible with the shapes
+    of K and y
+
+    :param regressor: sklearn-style regressor
+    :type regressor: object
+    :param K: kernel matrix with which to compare the
+        regression coefficients
+    :type K: array
+    :param y: target values with which to compare the
+        regression coefficients
+    :type y: array
+
+    """
+    if fitted_regressor.dual_coef_.ndim != y.ndim:
+        raise ValueError(
+            "The target regressor has a shape incompatible "
+            "with the supplied target space"
+        )
+    elif fitted_regressor.dual_coef_.ndim == 1:
+        if fitted_regressor.dual_coef_.shape[0] != K.shape[0]:
+            raise ValueError(
+                "The target regressor has a shape incompatible "
+                "with the supplied sample space"
+            )
+    else:
+        if fitted_regressor.dual_coef_.shape[0] != K.shape[0]:
+            raise ValueError(
+                "The target regressor has a shape incompatible "
+                "with the supplied sample space"
+            )
+        elif fitted_regressor.dual_coef_.shape[1] != y.shape[1]:
+            raise ValueError(
+                "The target regressor has a shape incompatible "
+                "with the supplied target space"
+            )
```

Comment:
Shouldn't this all be caught if you try and pass X to `regressor.predict`? I feel like you could avoid all of these raised errors by relying on the errors already thrown by sklearn.

```suggestion
    Checks that the coefficients of a fitted
    regression model is compatible with the shapes
    of X and y

    :param regressor: sklearn-style regressor
    :type regressor: object
    :param X: feature or kernel matrix with which to compare the
        regression coefficients
    :type X: array
    :param y: target values with which to compare the
        regression coefficients
    :type y: array

    """
    yp = fitted_regressor.predict(X)
    if yp.ndim != y.ndim or yp.shape[-1] != y.shape[-1]:
        raise ValueError("The regressor supplied predicts a quantity inconsistent with the shape of the supplied y.")
        
```

Reply:
You can use `self._validate_data(self, X, y)` from anything inheriting from `BaseEstimator` to check the dimensionality of X with respect to the coefficients. Checking `y` now.

Reply:
It doesn't look like there is a way to validate that the regressor outputs a `y` consistent with the supplied `y`, but the `self._validate_data(self, X, y)` _will_ flag an error if dim(y) is unexpected. 

btw all of this applies to kernel regressors as well, no need for separate functions. Btw, I would be happy if you just called `fitted_regressor._validate_data(X, y)` wherever you intended to call this function. That seems to be the implementation in sklearn.